### DISCUSSION TITLE
Is RAM Disk in warm restart additional memory?

### DISCUSSION THREAD

**Discussion by himanshusahu06** [memcached/memcached#1068] [General]
I came across https://github.com/memcached/memcached/wiki/WarmRestart which talks about persistent memory via DAX filesystem mounts. Let's assume I create 100GB instance with warm restart, does that mean I need 200 GB of RAM (100 for memcache and 100 for RAM disk)?

Also, is persistent memory (NVMEs) hard requirement for warm restart? Can we create RAM disk in **volatile memory**  (using tempfs instead of dax) and reuse the same to recover from cache flush (ofcourse, machine restart will flush the RAM but that's ok)?

**Comment by himanshusahu06** [comment]
@dormando

**Comment by dormando** [comment]
A DAX mount is a special filesystem for accessing "memory like" block devices, see: https://www.kernel.org/doc/Documentation/filesystems/dax.txt - When this is in use, memcached's item memory is directly accessed from that mount. There is no duplication. The hash table and various network buffers still exist in main RAM.

So you need 100GB of RAM disk space and then a couple gigabytes of RAM for hash table/whatever. Not 100/100.

Re: NVME's, no that's not necessary. Restartable mode works with tmpfs. It's primarily designed to use tmpfs, it just happens to also work with a dax mount.

**Comment by himanshusahu06** [comment]
Thanks! That was extremely helpful!
